django/test/client.py

```diff
@@ -482,7 +483,7 @@ def request(self, **request):
             if self.exc_info:
                 exc_info = self.exc_info
                 self.exc_info = None
-                raise exc_info[0](exc_info[1]).with_traceback(exc_info[2])
```

Comment:
@claudep, as far as I see, `raise exc_info[1]` works here. I'm not if `with_traceback(exc_info[2])` ever adds anything. Do you have a good understand of its purpose? 

Here's a sample traceback which is the same output as older versions of Django:
```
======================================================================
ERROR: test_detail_view_with_a_past_question (polls.tests.QuestionIndexDetailTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/mysite/polls/tests.py", line 128, in test_detail_view_with_a_past_question
    response = self.client.get(reverse('polls:detail', args=(past_question.id,)))
  File "/home/tim/code/django/django/test/client.py", line 519, in get
    response = super().get(path, data=data, secure=secure, **extra)
  File "/home/tim/code/django/django/test/client.py", line 333, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/home/tim/code/django/django/test/client.py", line 405, in generic
    return self.request(**r)
  File "/home/tim/code/django/django/test/client.py", line 486, in request
    raise exc_info[1]
  File "/home/tim/code/django/django/core/handlers/exception.py", line 35, in inner
    response = get_response(request)
  File "/home/tim/code/django/django/core/handlers/base.py", line 128, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/tim/code/django/django/core/handlers/base.py", line 126, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/tim/code/django/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/tim/code/django/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/tim/code/django/django/views/generic/detail.py", line 105, in get
    self.object = self.get_object()
  File "/home/tim/code/django/django/views/generic/detail.py", line 30, in get_object
    queryset = self.get_queryset()
  File "/home/tim/code/mysite/polls/views.py", line 42, in get_queryset
    raise CustomExceptionWith2PositionalParam('required', 'also_required')
polls.views.CustomExceptionWith2PositionalParam: ('required', 'also_required')
```
For the test that's added in this PR, the if statements in `six.reraise()` don't evaluate to True, so it does `raise value` (which is the same thing as `raise exc_info[1]`).
```
def reraise(tp, value, tb=None):
    if value is None:
        value = tp()
    if value.__traceback__ is not tb:
        raise value.with_traceback(tb)
    raise value
```

Reply:
Looking at [the C source code](https://github.com/python/cpython/blob/4d193bcc2560b824389e4d98d9d8b9b34e33dbaf/Objects/exceptions.c#L161-L173), `with_traceback(tb)` does `self.__traceback_ = tb`. The `with_traceback(...)` part of `exc_info[1].with_traceback(exc_info[2])` looks useless because  adding this assertion before that line always passes: `assert exc_info[1].__traceback__ is exc_info[2]`. I guess the `with_traceback()` addition was meant to emulate the `if value.__traceback__ is not tb: raise value.with_traceback(tb)` from `six.reraise()` but I don't see how that code could run.